### PR TITLE
[WFLY-12019] HostRemove should not remove services unless allow-resource-service-restart=true is set

### PR DESCRIPTION
**Pull request by gaol**
Issue: https://issues.redhat.com/browse/WFLY-12019

This PR tries to replace #16894 as that has test failures, and creating a new PR based on that will be efficient.

The test failures in #16894 is interacted  tests related, means that some other tests lead to a server state not cleaned, it is not easy to locate that test, but trying to reload the server when trying to modify the server can solve the test failure issue.

### CODE REVIEW

**Review by pferraro — CHANGES_REQUESTED** [review]
The fix looks good to me, my only comments are about the test itself.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/web/UndertowVirtualServerRemovalTestCase.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import static org.wildfly.extension.undertow.Constants.DEFAULT_SERVER;
+
+/**
+ * Tests on the virtual servers and hosts removal in undertow subsystem
+ *
+ * @author <a href="mailto:lgao@redhat.com">Lin Gao</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class UndertowVirtualServerRemovalTestCase extends ContainerResourceMgmtTestBase {
```
[the remark this answers is not in the document]
Comment (gaol) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/web/UndertowVirtualServerRemovalTestCase.java:58]: Thanks @pferraro for looking at this.

The test is trying to add a new server/host with some children resources, and remove it again during the test, and check the status after server/host removal. I don't know that we can test this within the subsystem unit test, I need to learn more before updating the tests if desirable. :)

[the remark this answers is not in the document]
Comment (pferraro) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/web/UndertowVirtualServerRemovalTestCase.java:58]: The KernelServices abstraction from the subsystem test framework has the ability to execute operations:
https://github.com/wildfly/wildfly-core/blob/main/subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/KernelServices.java#L44

**Review by darranl — CHANGES_REQUESTED** [review]
Is it possible to please have a more meaningful message for the test commit, five years in the future when someone comes along with git log or annotate to check the history of the test they end up with just a Jira ID they need to manually look up.

File: undertow/src/test/java/org/wildfly/extension/undertow/UndertowServerRemovalTestCase.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+ *  Tests server and host removal in Undertow subsystem.
+ *
+ * @author <a href="mailto:lgao@redhat.com">Lin Gao</a>
+ */
+public class UndertowServerRemovalTestCase extends AbstractUndertowSubsystemTestCase {
+
+    private static final String NODE_NAME = "node-name";
+
+    public UndertowServerRemovalTestCase() {
+        super(UndertowSubsystemSchema.VERSION_14_0);
```
Comment (pferraro) [inline undertow/src/test/java/org/wildfly/extension/undertow/UndertowServerRemovalTestCase.java:50]: Why not always use the current schema?
e.g. UndertowSubsystemSchema.CURRENT

File: undertow/src/test/java/org/wildfly/extension/undertow/UndertowServerRemovalTestCase.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    private PathAddress hostAddress(String server, String host) {
+        return serverAddress(server).append("host", host);
+    }
+
+    @Test
+    public void removeDefaultServerShouldFail() throws Exception {
+        KernelServices mainServices = null;
+        try {
+            final String defaultServer = "some-server";
+            mainServices = load("undertow-14.0.xml");
```
Comment (pferraro) [inline undertow/src/test/java/org/wildfly/extension/undertow/UndertowServerRemovalTestCase.java:82]: You can avoid hardcoding the version via:
this.load(this.getSubsystemXml());

**Comment by gaol** [comment]
@pferraro thanks Paul, I updated the test to avoid hard coding.

**Review by pferraro — APPROVED** [review]
Nice work!

**Pull request merged**